pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/IndexKey.java

```diff
@@ -46,7 +46,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof IndexKey)) {
```

Comment:
Any specific reason changing this? There are 2 flavors, one allows child class equal to the parent class, another doesn't. I think we usually perform the class check

Reply:
good to know!! will revert this.